dev/com.ibm.ws.session.cache_fat/test-applications/sessionCacheApp/src/session/cache/web/SessionCacheTestServlet.java

```diff
@@ -402,7 +408,24 @@ public void sessionGetTimeout(HttpServletRequest request, HttpServletResponse re
         long timeoutNS = TimeUnit.MINUTES.toNanos(1);
         for (long start = System.nanoTime(); cache.containsKey(sessionId) && System.nanoTime() - start < timeoutNS; TimeUnit.MILLISECONDS.sleep(500));
 
-        session.getAttribute(key);
+        String actual = (String) session.getAttribute(key);
+        assertEquals(expected, actual);
+    }
+
+    public void sessionPutTimeout(HttpServletRequest request, HttpServletResponse response) throws Throwable {
+        HttpSession session = request.getSession(true);
+        String key = request.getParameter("key");
+        String value = request.getParameter("value");
+        String sessionId = session.getId();
+        // poll for entry to be invalidated from cache
+        System.setProperty("hazelcast.config", InitialContext.doLookup("jcache/hazelcast.config")); // need to use same config file as server.xml
+        @SuppressWarnings("rawtypes")
+        Cache<String, ArrayList> cache = Caching.getCache("com.ibm.ws.session.meta.default_host%2FsessionCacheApp", String.class, ArrayList.class);
+        long timeoutNS = TimeUnit.MINUTES.toNanos(1);
```

Comment:
Is there a reason why the TIMEOUT_NS constant from the same class isn't used here?